drivers/pwm/pwm_nrfx.c

```diff
@@ -259,8 +264,12 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 			if (inverted) {
 				out_level ^= 1;
 			}
-
-			nrf_gpio_pin_write(psel, out_level);
+			if (pwm_is_fast(config)) {
```

Comment:
I would suggest to put a comment here why `fast` instance cannot utilize GPIO write:
```
Fast PWM instances output is connected directly to GPIO pads hence cannot be controlled by GPIO registers directly. Use 100% duty cycle playback instead.
```